docs/containers.md

```diff
@@ -14,6 +14,20 @@ Containers can have 2 or 3 dimensions, supporting e.g. tubes (2D), plates (2D) a
 
 When a sample is first created via the high-level API, it is not in a container. One can add it to one like this:
 
+TODO
+
+I think it could be helpful here to have an API that looks more like this:
+
+```
+a = Sample("A")
+plate = Container("c", rows=8, cols=12)
+plate[ContainerPosition(row="A", column="1")] = a
```

Comment:
Right. But this is only setting the dimensions of the `Container`. I.e. it's an 8x12 plate.

I had intentions for `plate["A1"]` (like we have in our extension framework) and similar to solve the case you're talking about.

Reply:
I'm not sure I follow. The problem I see is that I see the designation `A1` as ambiguous. I.e. I find it difficult to know for sure if `A` or `1` represents the row?

Reply:
Aha. The diff doesn't look like that's what being commented on, so it read like a misunderstanding.

But to comment on that suggestion then :) Sorry for how long it became, I hope it makes sense:

I believe it's much more common (if not a standard) that rows are named by letters and columns by numbers on plates. Googling "96 well plate" shows me only plates and diagrams using that convention, in cases where there are headers on it at all.

I think it should surely be an alternative to allow your suggestion (more on that below), but I personally wouldn't want to have to type that if the other is a convention as then it's more of a low level API than a high level one, and we should in my mind aim for a domain driven design and readability for the high level API. In our LIMS extension framework, we allow both.

It would also make even this document more hard to read for those that are accustomed to this convention, for `"A1" => "B1"` is so much easier to read than `ContainerPosition(row="A", column="1") => ContainerPosition(row="A", column="1")`.

While the latter is in some sense clearer, it's in the way if one is accustomed to the convention. It's similar to how we would never require one to say what are row and what are column indexes while using `i` and `j` to address locations in a matrix.

So, back to the implementation I had visioned. I was planning on an approach where the plate generally expects some kind of `Index` object, but also an "index-like" object. So given a string, it would try to parse it to an index object given the most common convention for that kind of container. Does that make sense?

Reply:
I think it makes sense if that is a strongly rooted convention in the domain to go with that, and I agree, that once you understand it `A1` is easier to read than `ContainerPosition(row='B', column='1')`. I like the "index-like" object idea. Thanks for taking the time to explain your reasoning. :+1: